maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultPluginValidationManager.java

```diff
@@ -0,0 +1,248 @@
+                    MAVEN_PLUGIN_VALIDATION_KEY,
+                    level,
+                    Arrays.toString(ValidationLevel.values()));
+            return ValidationLevel.DEFAULT;
+        }
+    }
+
+    private String pluginKey(String groupId, String artifactId, String version) {
+        return groupId + ":" + artifactId + ":" + version;
+    }
+
+    private String pluginKey(MojoDescriptor mojoDescriptor) {
+        PluginDescriptor pd = mojoDescriptor.getPluginDescriptor();
+        return pluginKey(pd.getGroupId(), pd.getArtifactId(), pd.getVersion());
+    }
+
+    private String pluginKey(Artifact pluginArtifact) {
+        return pluginKey(pluginArtifact.getGroupId(), pluginArtifact.getArtifactId(), pluginArtifact.getVersion());
+    }
+
+    @Override
+    public void reportPluginValidationIssue(RepositorySystemSession session, Artifact pluginArtifact, String issue) {
+        String pluginKey = pluginKey(pluginArtifact);
+        PluginValidationIssues pluginIssues =
+                pluginIssues(session).computeIfAbsent(pluginKey, k -> new PluginValidationIssues());
+        pluginIssues.reportPluginIssue(null, null, issue);
+    }
+
+    @Override
+    public void reportPluginValidationIssue(MavenSession mavenSession, MojoDescriptor mojoDescriptor, String issue) {
+        String pluginKey = pluginKey(mojoDescriptor);
+        PluginValidationIssues pluginIssues = pluginIssues(mavenSession.getRepositorySession())
+                .computeIfAbsent(pluginKey, k -> new PluginValidationIssues());
+        pluginIssues.reportPluginIssue(pluginDeclaration(mojoDescriptor), pluginOccurrence(mavenSession), issue);
+    }
+
+    @Override
+    public void reportPluginMojoValidationIssue(
+            MavenSession mavenSession, MojoDescriptor mojoDescriptor, Class<?> mojoClass, String issue) {
+        String pluginKey = pluginKey(mojoDescriptor);
+        PluginValidationIssues pluginIssues = pluginIssues(mavenSession.getRepositorySession())
+                .computeIfAbsent(pluginKey, k -> new PluginValidationIssues());
+        pluginIssues.reportPluginMojoIssue(
+                pluginDeclaration(mojoDescriptor),
+                pluginOccurrence(mavenSession),
+                mojoInfo(mojoDescriptor, mojoClass),
+                issue);
+    }
+
+    private void reportSessionCollectedValidationIssues(MavenSession mavenSession) {
+        ValidationLevel validationLevel = validationLevel(mavenSession.getRepositorySession());
+        ConcurrentHashMap<String, PluginValidationIssues> issuesMap = pluginIssues(mavenSession.getRepositorySession());
+        if (!issuesMap.isEmpty()) {
+
+            logger.warn("");
+            logger.warn("Plugin validation issues were detected in {} plugin(s)", issuesMap.size());
+            logger.warn("");
+            if (validationLevel == ValidationLevel.DISABLED || !logger.isWarnEnabled()) {
+                return;
+            }
```

Comment:
If disable nothing should be logged

Reply:
No, I think that the fact there were issues still should be logged. Otherwise it would go unnoticed.

Reply:
I mean, in 3.9.0 or 3.9.1 there is no possibility to totally "mute" validation, is there?